eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/serialization/commons/Extra.java

```diff
@@ -0,0 +1,40 @@
+package com.schedjoules.eventdiscovery.framework.serialization.commons;
+
+import android.content.Intent;
+import android.os.Bundle;
+
+import com.schedjoules.eventdiscovery.framework.serialization.Keys;
+import com.schedjoules.eventdiscovery.framework.serialization.core.Key;
+import com.schedjoules.eventdiscovery.framework.utils.factory.Factory;
+import com.schedjoules.eventdiscovery.framework.utils.optionals.AbstractCachingOptional;
+
+import org.dmfs.optional.Absent;
+import org.dmfs.optional.Optional;
+
+
+/**
+ * @author Gabor Keszthelyi
+ */
+public final class Extra<T> extends AbstractCachingOptional<T>
+{
+
+    public Extra(final Key<T> key, final Intent intent)
+    {
+        super(new Factory<Optional<T>>()
+        {
+            @Override
+            public Optional<T> create()
+            {
+                if (intent != null)
```

Comment:
Right, simplified it and also removed the `null` check for `intent` - you're right that can never really be `null` normally.